### PR TITLE
Growth API endpoints

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
added `/network/direct-staked-info` and `/network/delegated-info`

### CODE REVIEW

File: api/groups/baseNetworkGroup.go [hunk 1]
```diff
@@ -76,6 +78,28 @@ func (group *networkGroup) getEconomicsData(c *gin.Context) {
 	c.JSON(http.StatusOK, economicsData)
 }
 
+// getEconomicsData will expose the economics data metrics from an observer (if any available) in json format
```
Comment (miiu96) [inline api/groups/baseNetworkGroup.go:81]: change comment

File: api/groups/baseNetworkGroup.go [hunk 2]
```diff
@@ -76,6 +78,28 @@ func (group *networkGroup) getEconomicsData(c *gin.Context) {
...
+	directStakedInfo, err := group.facade.GetDirectStakedInfo()
+	if err != nil {
+		shared.RespondWith(c, http.StatusInternalServerError, nil, err.Error(), data.ReturnCodeInternalError)
+		return
+	}
+
+	c.JSON(http.StatusOK, directStakedInfo)
+}
+
+// getEconomicsData will expose the economics data metrics from an observer (if any available) in json format
```
Comment (miiu96) [inline api/groups/baseNetworkGroup.go:92]: change comment

File: process/nodeStatusProcessor.go [hunk 3]
```diff
@@ -24,6 +24,12 @@ const NodeStatusPath = "/node/status"
 // NodeStatusPath represents the path where an observer exposes all the issued ESDTs
 const AllIssuedESDTsPath = "/network/esdts"
```
Comment (miiu96) [inline process/nodeStatusProcessor.go:25]: group all const (
...
...
)

File: process/nodeStatusProcessor.go [hunk 4]
```diff
@@ -126,6 +132,54 @@ func (nsp *NodeStatusProcessor) GetAllIssuedESDTs() (*data.GenericAPIResponse, e
...
+func (nsp *NodeStatusProcessor) GetDelegatedInfo() (*data.GenericAPIResponse, error) {
+	observers, err := nsp.proc.GetObservers(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		var delegatedInfoResponse *data.GenericAPIResponse
+
+		_, err := nsp.proc.CallGetRestEndPoint(observer.Address, DelegatedInfoPath, &delegatedInfoResponse)
```
Comment (miiu96) [inline process/nodeStatusProcessor.go:145]: maybe you can add a cache for these new endpoints.

File: process/nodeStatusProcessor.go [hunk 5]
```diff
@@ -126,6 +132,54 @@ func (nsp *NodeStatusProcessor) GetAllIssuedESDTs() (*data.GenericAPIResponse, e
...
+		if err != nil {
+			log.Error("network delegated info request", "observer", observer.Address, "error", err.Error())
+			continue
+		}
+
+		log.Info("network delegated info request", "shard id", observer.ShardId, "observer", observer.Address)
```
Comment (miiu96) [inline process/nodeStatusProcessor.go:151]: shard ID

File: process/nodeStatusProcessor.go [hunk 6]
```diff
@@ -126,6 +132,54 @@ func (nsp *NodeStatusProcessor) GetAllIssuedESDTs() (*data.GenericAPIResponse, e
...
+	for _, observer := range observers {
+		var directStakedResponse *data.GenericAPIResponse
+
+		_, err := nsp.proc.CallGetRestEndPoint(observer.Address, DirectStakedPath, &directStakedResponse)
+		if err != nil {
+			log.Error("network direct staked request", "observer", observer.Address, "error", err.Error())
+			continue
+		}
+
+		log.Info("network direct staked request", "shard id", observer.ShardId, "observer", observer.Address)
```
Comment (miiu96) [inline process/nodeStatusProcessor.go:175]: shard ID

File: api/groups/baseNetworkGroup.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/groups/baseNetworkGroup.go:81]: done

File: api/groups/baseNetworkGroup.go (hunk 2 above)
Comment (bogdan-rosianu) [reply api/groups/baseNetworkGroup.go:92]: done

File: process/nodeStatusProcessor.go (hunk 3 above)
Comment (bogdan-rosianu) [reply process/nodeStatusProcessor.go]: done

File: process/nodeStatusProcessor.go (hunk 4 above)
Comment (bogdan-rosianu) [reply process/nodeStatusProcessor.go:147]: I don't think it's not needed since they are secured

File: process/nodeStatusProcessor.go (hunk 5 above)
Comment (bogdan-rosianu) [reply process/nodeStatusProcessor.go]: done

File: process/nodeStatusProcessor.go (hunk 6 above)
Comment (bogdan-rosianu) [reply process/nodeStatusProcessor.go]: done

**Review by iulianpascalau — DISMISSED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**